File: Java/23_Math/LC_2591_DistributeMoneyToMaximumChildren.java
class Solution {
    public int distMoney(int money, int children) {

        if (money < children) {
            return -1;
        }

        if (8 * (children - 1) + 4 == money) {
            return children - 2;
        }

        if (money > children * 8) {
            return children - 1;
        }

        return (money - children) / 7;
    }
}

public class LC_2591_DistributeMoneyToMaximumChildren {

}

File: JavaScript/23_Math/LC_2591_DistributeMoneyToMaximumChildren.js
/**
 * @param {number} money
 * @param {number} children
 * @return {number}
 */
var distMoney = function(money, children) {
    if(money < children) return-1;
    if(money > children*8) return children-1;//每个人都分8，有剩余全部给一个人 
    if(money === (children*8-4)) return children-2;//n-1分8，正好剩下4，要调整2个人
    return Math.floor((money-children)/7);
 };